router-tests/safelist_test.go

```diff
@@ -0,0 +1,166 @@
+package integration
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/wundergraph/cosmo/router-tests/testenv"
+	"github.com/wundergraph/cosmo/router/pkg/config"
+	"go.uber.org/zap/zapcore"
+	"net/http"
+	"testing"
+)
+
+func TestSafelist(t *testing.T) {
+	t.Parallel()
+
+	var (
+		persistedQuery        = "query Employees {\n employees {\n id\n }\n}"
+		nonPersistedQuery     = "query Employees {\n\n\n employees {\n id\n }\n}"
+		queryWithDetails      = "query Employees {\n employees {\n id\n details {\n forename\n} \n}\n}"
+		persistedNotFoundResp = `{"errors":[{"message":"PersistedQueryNotFound","extensions":{"code":"PERSISTED_QUERY_NOT_FOUND"}}]}`
+	)
+
+	t.Run("router fails if APQ and Safelist are both enabled", func(t *testing.T) {
+		testenv.FailsOnStartup(t, &testenv.Config{
+			ApqConfig: config.AutomaticPersistedQueriesConfig{Enabled: true},
+			ModifySecurityConfiguration: func(securityConfiguration *config.SecurityConfiguration) {
+				securityConfiguration.Safelist = config.EnableOperationConfiguration{Enabled: true}
```

Comment:
Will safelist affect cache warming? Maybe we should add one test there as well to see if it still works as intended.